pkg/controller/operators/catalog/operator.go

```diff
@@ -976,22 +976,23 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		// not-satisfiable error
 		if _, ok := err.(solver.NotSatisfiable); ok {
 			logger.WithError(err).Debug("resolution failed")
-			updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
+			_, updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
 			if updateErr != nil {
 				logger.WithError(updateErr).Debug("failed to update subs conditions")
 			}
 			return nil
 		}
-		updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
+		_, updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
 		if updateErr != nil {
 			logger.WithError(updateErr).Debug("failed to update subs conditions")
 		}
 		return err
 	} else {
-		updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "", "", false)
+		subs, updateErr := o.setSubsCond(updatedSubs, v1alpha1.SubscriptionResolutionFailed, "", "", false)
```

Comment:
Why does this only reset `updatedSubs`?

Reply:
@benluddy do you mean why are the other calls to `setSubsCond` also not using `updatedSubs`?

Reply:
No -- if only some of the Subscriptions in the namespace are present in `updatedSubs`, what is going to clear this status condition for the others?

Reply:
I see. This is probably highlighting a design choice we made, i.e the decision to project this to all subscriptions in the namespace. 

The point of this fix is that inside this `else` block, we work with the updated subs so that we can set the updated subs with ResolutionFailed condition reset to false, and that is the list of subs that the rest of the method continues with. 

If we pass in the full list of `subs` instead  of just the `updatedSubs`, we get back the list of all the subs, that is then overriden [here](https://github.com/operator-framework/operator-lifecycle-manager/pull/2296/files#diff-a1760d9b7ac1e93734eea675d8d8938c96a50e995434b163c6f77c91bace9990R995), i.e we have the wrong set of `updatedSubs` at that point.

Reply:
Possible solutions include: 
1. We change the decision to project to all subscriptions in the namespace, and instead project the condition to only those subscriptions that are involved in the resolution step. 
2. We add extra logic to list all the subscriptions in the namespace after the `updatedSubs` are updated, and call the `setSubCond` function again. With the introduction of the check inside `setSubCond` to skip subscriptions that already have the desired state, we can be sure we're not doing the same work twice.

Reply:
Updated PR with option 2.